File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/di/component/RepositoryComponent.java
package com.kalashnyk.denys.gmapsapp.di.component;

import com.kalashnyk.denys.gmapsapp.di.module.RepositoryModule;
import com.kalashnyk.denys.gmapsapp.di.scope.RepositoryScope;
import com.kalashnyk.denys.gmapsapp.repository.AppRepository;

import dagger.Component;

@RepositoryScope
@Component(modules = {RepositoryModule.class}, dependencies = {ApiComponent.class, DaoComponent.class})
public interface RepositoryComponent {
    AppRepository getRepository();
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/di/component/RepositoryComponent.kt
package com.kalashnyk.denys.kotlinsample.di.component

import com.kalashnyk.denys.kotlinsample.di.module.RepositoryModule
import com.kalashnyk.denys.kotlinsample.di.scope.RepositoryScope
import com.kalashnyk.denys.kotlinsample.repository.AppRepository
import dagger.Component

@RepositoryScope
@Component(modules = [RepositoryModule::class], dependencies = [ApiComponent::class, DatabaseComponent::class])
interface RepositoryComponent {
    val repository: AppRepository
}